go/ai/generate.go

```diff
@@ -1257,3 +1342,152 @@ func executeResourcePart(ctx context.Context, r api.Registry, resourceURI string
 
 	return output.Content, nil
 }
+
+// NewBackgroundModel defines a new model that runs in the background
+func NewBackgroundModel(
+	name string,
+	opts *BackgroundModelOptions,
+) *BackgroundModel {
+	return &BackgroundModel{core.NewBackgroundAction[*ModelRequest, *ModelResponse](name, nil,
+		opts.Start, opts.Check, opts.Cancel)}
+}
+
+// DefineBackgroundModel defines and registers a new model that runs in the background.
+func DefineBackgroundModel(
+	r *registry.Registry,
+	name string,
+	opts *BackgroundModelOptions,
+) *BackgroundModel {
+	m := NewBackgroundModel(name, opts)
+	// Use the merged metadata from NewBackgroundModel
+	mergedMetadata := make(map[string]any)
+	if opts.Metadata != nil {
+		for k, v := range opts.Metadata {
+			mergedMetadata[k] = v
+		}
+	}
+
+	// Add model-specific metadata
+	label := opts.Label
+	if label == "" {
+		label = name
+	}
+	mergedMetadata["model"] = map[string]any{
+		"label":    label,
+		"versions": opts.Versions,
+		"supports": opts.Supports,
+	}
+	if opts.ConfigSchema != nil {
+		mergedMetadata["customOptions"] = opts.ConfigSchema
+		if modelMeta, ok := mergedMetadata["model"].(map[string]any); ok {
+			modelMeta["customOptions"] = opts.ConfigSchema
+		}
+	}
```

Comment:
This should all be done in `NewBackgroundModel` and this function should just register the action. Check `NewModel` and `DefineModel`.